api/nodes/serializers.py

```diff
@@ -127,10 +127,15 @@ class NodeSerializer(JSONAPISerializer):
     )
 
     forked_from = RelationshipField(
-        related_view='nodes:node-detail',
+        related_view=lambda n: 'registrations:registration-detail' if getattr(n, 'is_registration', False) else 'nodes:node-detail',
```

Comment:
:) @chrisseto helped me with implement this elsewhere awhile back.  Remember how `node` and `original_node` can be nodes or registrations on logs?  So we used this and then I put a check in the RelationshipField to see if the view is a callable and then fetch the node-detail or registration-detail view.